### PR TITLE
Fixes, optimizations and additions to WiFi

### PR DESCRIPTION
**Pull request by Eloquence4**

### CODE REVIEW

**Comment by nkolban** [comment]
I'm seeing a series of additions to these pull-requests ... do you want me to hold off merging them?

File: cpp_utils/WiFi.cpp [hunk 1]
```diff
@@ -42,13 +42,27 @@ static void setDNSServer(char *ip) {
...
+    , wifiEventHandler(nullptr)
+{
     wifiEventHandler = new WiFiEventHandler();
 }
 
+/**
+ * @brief Deletes the event handler that was used by the class
+ */
+WiFi::~WiFi() {
+    delete wifiEventHandler;
```
Comment (nkolban) [inline cpp_utils/WiFi.cpp]: Do we need to check that wifiEventHandler isn't null before deleting?

Comment (Eloquence4) [reply cpp_utils/WiFi.cpp]: No, C++ does that automatically. If it's null delete just does nothing, so the if is irrelevant.

**Comment by Eloquence4** [comment]
Nah, code is stable. I make sure my code is stable before i make a commit. You can merge if you want.

**Pull request merged**